services/src/test/java/io/scalecube/services/TransportSenderTest.java

```diff
@@ -0,0 +1,67 @@
+package io.scalecube.services;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotEquals;
+
+import io.scalecube.transport.Message;
+import io.scalecube.transport.Transport;
+
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class TransportSenderTest {
+
+  @Test
+  public void test_transport_sender() throws InterruptedException {
+
+    Transport transport = Transport.bindAwait();
+    TransportSender sender = new TransportSender(transport);
+
+    assertEquals(transport.address(), sender.address());
+
+    CountDownLatch latch = new CountDownLatch(1);
+    sender.listen().subscribe(onNext -> {
+      latch.countDown();
+    });
+
+    latch.await(1, TimeUnit.SECONDS);
+    sender.send(transport.address(), Message.builder().data("ping").build());
```

Comment:
This won't help since you are waiting for latch before you send the message, also to make any sense out of it, you actually need to check that latch has reached zero...

Also don't forget to shutdown transport at the end of the test (generally I recommend to do it at the finally block)